src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+
+                    // Read the next charLength characters
+                    buffer = new char[v2Region.CharLength];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+                else // Use charOffset
+                {
+                    // Read the next charLength characters
+                    char[] buffer = new char[v2Region.CharOffset];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private int GetRegionEndColumn(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+                string sourceLine = null;
+
+                // Read down to the line before endLine
+                for (int i = 1; i < v2Region.EndLine; sourceLine = reader.ReadLine(), i++) { }
+                result = sourceLine.Length;
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private TextReader GetFileTextReader(Uri uri, out Encoding encoding)
+        {
+            TextReader reader = null;
+            encoding = null;
+            var failureReason = new StringBuilder();
+
+            if (uri != null && _currentV2Run.Files != null)
+            {
+                FileData fileData;
+                if (_currentV2Run.Files.TryGetValue(uri.OriginalString, out fileData))
+                {
+                    // Determine the encoding
+                    string encodingName = null;
+
+                    if (fileData.Contents?.Text != null)
+                    {
+                        // Embedded text shall be UTF-8 encoded
+                        encoding = Encoding.UTF8;
+                    }
+                    else
+                    {
+                        encodingName = fileData.Encoding ?? _currentV2Run.DefaultFileEncoding;
+                        encoding = GetFileEncoding(encodingName);
+                    }
+
+                    if (encoding != null)
+                    {
+                        try
+                        {
+                            if (fileData.Contents != null)
+                            {
+                                // Embedded text file content
+                                string content = fileData.Contents.Text ?? SarifUtilities.DecodeBase64String(fileData.Contents.Binary, encoding);
```

Comment:
They might both be null. The condition on Line 650 should be:
```
if (fileData.Contents != null && (fileData.Contents.Text != null || fileData.Contents.Binary != null))
```

Reply:
Fixed

Reply:
fixed